File: benchmarks/role4j/src/main/java/bank/Account.java
package bank;

import net.role4j.IPlayer;

public class Account implements IPlayer {
  private float balance;

  public Account() {

  }

  public void setBalance(float amount) {
    balance = amount;
  }

  public void decrease(float amount) {
    //System.out.println("decrease");
    balance -= amount;
  }

  public void increase(float amount) {
    //System.out.println("increase");
    balance += amount;
  }

  public float getBalance() {
    return balance;
  }
}

File: benchmarks/scroll/src/main/scala/bank/Account.scala
package bank

trait Accountable

trait Decreasable extends Accountable {
  def decrease(amount: Float): Unit
}

trait Increasable extends Accountable {
  def increase(amount: Float): Unit
}

case class Account(var balance: Float = 0.0f) extends Increasable with Decreasable {

  def increase(amount: Float): Unit =
    balance = balance + amount

  def decrease(amount: Float): Unit =
    balance = balance - amount

  def getBalance(): Float = balance
}
